chain/messagepool/messagepool.go

```diff
@@ -447,6 +447,8 @@ func New(ctx context.Context, api Provider, ds dtypes.MetadataDS, us stmgr.Upgra
 	return mp, nil
 }
 
+// TryForEachPendingMessage calls the provided function for each pending message in the pool, if possible.
+// Returns an error if the lock cannot be acquired or any f call fails
 func (mp *MessagePool) TryForEachPendingMessage(f func(cid.Cid) error) error {
```

Comment:
It would be easy to make that change - but i'm not sure if it's necessary or an improvement - from the Range documentation `Range does not block other methods on the receiver` https://pkg.go.dev/sync#Map.Range

Reply:
Oh, if it doesn't block other methods we can probably just drop the lock